### PR TITLE
Preserve full URL when using 'anchor' function.

### PR DESCRIPTION
**Pull request by nikuda**
At the moment the url is replaced with hash, which has a negative impact on usability - i.e. you're not able to ctrl+click to open up a link in a new tab etc. With this patch the full url is rendered, while still being intercepted by mercury-router so you don't get an app-reload by clicking them directly.
- Moved the more complete example from `lib/router.js` into readme.md. And removed the old one.
- Simplified testing and added test for new anchor behaviour.

### CODE REVIEW

**Comment by nikuda** [comment]
@Raynos do you see any potential issues with this? The meat of the change is this a315fee804a0b158ade7800b8ecb5fa2a09bc5c0.

File: lib/route-anchor.js [hunk 1]
```diff
@@ -20,6 +17,6 @@ function anchor(props, text) {
     return h('a', props, text);
 
     function pushState() {
-        routeAtom.set(href);
+        routeAtom.set(props.href);
```
Comment (mynameistechno) [inline lib/route-anchor.js:20]: :+1:

**Comment by nikuda** [comment]
@twilson63, fwiw - this patch is ok we've been using this for the last couple of weeks without issues.

**Comment by twilson63** [comment]
merged

**Pull request merged**